training/training_job.js

```diff
@@ -398,67 +352,60 @@ module.exports = class Job {
             this.child.kill('SIGTERM');
     }
 
-    fail(error) {
+    async fail(error) {
         if (this.data.status !== 'queued' && !this._killed) {
             console.error(`Job ${this.data.id} failed during task ${this._currentTaskName()}: ${error}`);
             if (error.stack)
                 console.error(error.stack);
         }
         this.data.status = 'error';
         this.data.error = error.message;
-        this.complete();
```

Comment:
should we update db on failure here? Or it's already been done somewhere else?

Reply:
`_notifyJobComplete` will run a single transaction and update everything, including all dependent jobs